bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/types/ArrayType.java

```diff
@@ -28,7 +28,7 @@
  * @since 2.0.0
  */
 @SuppressWarnings("unchecked")
-public interface ArrayType extends Type {
+public interface ArrayType extends Type, SelectivelyImmutableReferenceType {
```

Comment:
```suggestion
public interface ArrayType extends SelectivelyImmutableReferenceType {
```

Few other applicable places too.